app/controllers/api/stateless/braintree/payment_methods_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Api
+  module Stateless
+    module Braintree
+      class PaymentMethodsController < StatelessController
+        before_filter :authenticate_request!
+
+        def index
+          @payment_methods = PaymentHelper::Braintree.payment_methods_for_member(@current_member)
+        end
+
+        def destroy
+          @payment_method = PaymentHelper::Braintree.payment_method_for_member(member: @current_member, id: params[:id])
+
+          result = begin
+                     ::Braintree::PaymentMethod.delete(@payment_method.token)
+                   rescue ::Braintree::NotFoundError
+                     @payment_method.destroy
+                   end
+
+          @payment_method.destroy if result.success?
```

Comment:
OK, I think I understand. I think what confused me was setting `result` to the return value of `@payment_method.destroy`, since `@payment_method.destroy.success?` would (I think) raise a `NoMethodError`. What if it looked more like -
```
begin
  result = ::Braintree::PaymentMethod.delete(@payment_method.token)
  @payment_method.destroy if result.success?
rescue ::Braintree::NotFoundError
  @payment_method.destroy
end
```